cmake/Embed.cmake

```diff
@@ -21,17 +21,21 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 #####################################################################################
-find_program(EMBED_LD ld)
-find_program(EMBED_OBJCOPY objcopy)
 
-option(EMBED_USE_LD "Use ld to embed data files" OFF)
```

Comment:
We shouldn't remove this option.

Reply:
The technique with LD on Linux is the same as RC on Windows. I changed `EMBED_USE_LD` to `EMBED_USE_RESOURCES` because, in both cases, the literature refers to both techniques as embedding resource files in program binary. Here is the example: https://atwillys.de/content/cc/embedding-resource-files-in-a-c-plus-plus-program-binary-on-linux-unix/?lang=en. I can easily find tonnes more. We only need one option to control resources or `*.cpp` files. It makes no sense to have both and expose `EMBED_USE_LD` on Windows and `EMBOED_USE_RC` on Linux.

Reply:
> The technique with LD on Linux is the same as RC on Windows. I changed EMBED_USE_LD to EMBED_USE_RESOURCES because, in both cases, the literature refers to both techniques as embedding resource files in program binary.

The name is the technique used for embedding the resource. We are embedding the resource when we use the C arrays as well, so the `EMBED_USE_RESOURCES` flag would apply there as well, so it doesnt make sense.

> It makes no sense to have both and expose EMBED_USE_LD on Windows and EMBOED_USE_RC on Linux.

I would prefer the two flags because there are other techniques to embed resources as well(such as using `#embed`) so I would prefer to have a flag to specify the technique being used to embed the resources.

Reply:
I prefer a single option for LD and RC (e.g., `EMBED_USE_BINARY` or `EMBED_USE_NATIVE` or `EMBED_USE_OS_NATIVE`) and separate for `#embed` when we switch to C++23 (e.g. `EMBED_USE_CXX_EMBED` or `EMBED_USE_CXX` or `EMBED_USE_CXX_NATIVE`) or something like that. Two flags for LD and RC would result in different Linux and Windows building instructions. One cannot use LD on Windows and RC on Linux. I prefer to keep them the same for both OSes.

Reply:
We should probably have one `EMBED_USE` variable that can be set to `EMBED_USE=RC` for windows resource files, `EMBED_USE=LD` for gnu linker, and `EMBED_USE=CArrays` for the carrays version.

We can also use `set_property(CACHE EMBED_USE PROPERTY STRINGS "LD;CArrays")` to set the options for linux, so in cmake gui(or tui) you can select the option you want.

Reply:
Done